src/test/java/space/obminyashka/items_exchange/service/RefreshTokenServiceTest.java

```diff
@@ -1,55 +1,125 @@
+
+        assertAll(
+                () -> assertEquals(EXPECTED_REFRESH_TOKEN, actualRefreshToken.getToken()),
+                () -> verify(jwtTokenService).generateRefreshToken(EXPECTED_USERNAME),
+                () -> verify(jwtTokenService).generateRefreshTokenExpirationTime(),
+                () -> verify(refreshTokenRepository)
+                        .createRefreshToken(eq(EXPECTED_USERNAME), eq(EXPECTED_REFRESH_TOKEN), any())
+        );
+    }
+
+    @Test
+    void createRefreshToken_whenRefreshTokenExistsInDB_shouldUpdateToken() {
+        var expectedRefreshToken = new RefreshToken().setToken("");
+        mockJwtRefreshTokenData(expectedRefreshToken.getToken());
+
+        final var actualRefreshToken = refreshTokenService
+                .createRefreshToken(expectedRefreshToken.getToken(), EXPECTED_USERNAME);
+
+        assertAll(
+                () -> assertEquals(expectedRefreshToken.getToken(), actualRefreshToken.getToken()),
+                () -> verify(jwtTokenService).generateRefreshToken(EXPECTED_USERNAME),
+                () -> verify(jwtTokenService).generateRefreshTokenExpirationTime(),
+                () -> verify(refreshTokenRepository)
+                        .updateRefreshToken(eq(EXPECTED_USERNAME), eq(expectedRefreshToken.getToken()), any())
+        );
+    }
+
+    private void mockJwtRefreshTokenData(String refreshToken) {
+        when(jwtTokenService.generateRefreshToken(anyString())).thenReturn(refreshToken);
         when(jwtTokenService.generateRefreshTokenExpirationTime()).thenReturn(LocalDateTime.MAX);
-        when(refreshTokenRepository.save(any())).thenAnswer(invocation -> invocation.getArgument(0));
+    }
 
-        final var refreshToken = refreshTokenService.createRefreshToken(testUser);
+    @Test
+    void renewAccessTokenByRefreshToken_whenAccessTokenExpires_shouldRenewAccessToken() {
+        var existRefreshToken = createRefreshToken(1);
+        when(refreshTokenRepository.findByToken(any())).thenReturn(Optional.of(existRefreshToken));
+        when(jwtTokenService.createAccessToken(anyString(), any(Role.class))).thenReturn(EXPECTED_ACCESS_TOKEN);
+
+        var actualAccessToken = refreshTokenService.renewAccessTokenByRefresh(existRefreshToken.getToken());
+
+        assertAll(
+                () -> assertTrue(actualAccessToken.isPresent()),
+                () -> assertEquals(Optional.of(EXPECTED_ACCESS_TOKEN), actualAccessToken),
+                () -> verify(refreshTokenRepository).findByToken(existRefreshToken.getToken()),
+                () -> verify(jwtTokenService)
+                        .createAccessToken(EXPECTED_USERNAME, existRefreshToken.getUser().getRole())
+        );
+    }
+
+    @Test
+    void renewAccessTokenByRefreshToken_whenAccessTokenDoesNotExpiry_shouldNotRenewAccessToken() {
+        var existRefreshToken = createRefreshToken(-1);
```

Comment:
Another "magic number".
If you're willing to keep that you should give a better name to the varible